odoo/fields.py

```diff
@@ -1654,7 +1654,7 @@ def convert_to_column(self, value, record, values=None):
         # unicode in some circumstances, hence the str() cast here.
         # This str() coercion will only work for pure ASCII unicode strings,
         # on purpose - non base64 data must be passed as a 8bit byte strings.
-        if not value:
+        if value == False:
```

Comment:
It might be better to change the order of the tests - test for bytes first, and then test for false-yness? That way we do the right thing with `b""` and don't need to change `not value` to `value == False` (or the technically more correct `value is False`).  If we were to keep the test for `False`, what would happen if `None` were passed in (if that's possible)?

Reply:
I'd forgotten about `None` entirely, thanks! The comment above seems to suggest that `str`s go through here as well, so I had to add a test for those when moving the bytes test up.